src/codeql.ts

```diff
@@ -616,20 +621,52 @@ export async function getCodeQLForCmd(
         extraArgs.push("--no-sublanguage-file-coverage");
       }
 
-      await runTool(
-        cmd,
-        [
-          "database",
-          "init",
-          "--db-cluster",
-          config.dbLocation,
-          `--source-root=${sourceRoot}`,
-          ...(await getLanguageAliasingArguments(this)),
-          ...extraArgs,
-          ...getExtraOptionsFromEnv(["database", "init"]),
-        ],
-        { stdin: externalRepositoryToken },
-      );
+      try {
+        await runTool(
+          cmd,
+          [
+            "database",
+            "init",
+            "--db-cluster",
+            config.dbLocation,
+            `--source-root=${sourceRoot}`,
+            ...(await getLanguageAliasingArguments(this)),
+            ...extraArgs,
+            ...getExtraOptionsFromEnv(["database", "init"]),
+          ],
+          { stdin: externalRepositoryToken },
+        );
+      } catch (e) {
+        if (e instanceof CommandInvocationError) {
+          if (isCliConfigurationError(CliError.InitCalledTwice, e.message)) {
+            throw new util.UserError(
+              processCliConfigurationError(CliError.InitCalledTwice, e.message),
+            );
+          }
+          if (
+            isCliConfigurationError(
+              CliError.IncompatibleWithActionVersion,
+              e.message,
+            )
+          ) {
+            throw new util.UserError(
+              processCliConfigurationError(
+                CliError.IncompatibleWithActionVersion,
+                e.message,
+              ),
+            );
+          }
+          if (isCliConfigurationError(CliError.InvalidSourceRoot, e.message)) {
+            throw new util.UserError(
+              processCliConfigurationError(
+                CliError.InvalidSourceRoot,
+                e.message,
+              ),
+            );
+          }
+        }
+        throw e;
```

Comment:
Sorry, I definitely wasn't sufficiently clear in my original (late night) comment, but I was hoping we could end up with something like this...
```suggestion
throw processCliConfigurationError(error);
```
So we'd end up not having the export the error types at all, just the processing function that would both identify and then enhance/change the error as required. Does that make sense? (quite possibly not... 😅)

Reply:
Actually, it was clear! But I ended up not really being able to do that because of the `NoJavaScriptTypeScriptCodeFound` case, which also requires a check to the CLI version 🤔 it seemed a little extraneous to _also_ add that case and logic to the configuration file.

It seems like that specific error can be removed once we stop supporting CLI v2.11.6, which is our oldest supported version now, so maybe we can make that refactoring change once it's deprecated.

For now — do you think it was cleaner before? 😆

Reply:
Ah, I see... 🤔

I think you might be right that it was clearer beforehand, apologies for derailing this PR somewhat! :laugh